js/piwik.js

```diff
@@ -4062,81 +4058,6 @@ if (typeof window.Piwik !== 'object') {
                 return false;
             }
 
-            function replaceHrefIfInternalLink(contentBlock)
-            {
-                if (!contentBlock) {
-                    return false;
-                }
-
-                var targetNode = content.findTargetNode(contentBlock);
-
-                if (!targetNode || content.shouldIgnoreInteraction(targetNode)) {
-                    return false;
-                }
-
-                var link = getLinkIfShouldBeProcessed(targetNode);
-
-                if (linkTrackingEnabled && link && link.type) {
-
-                    return false; // will be handled via outlink or download.
-                }
-
-                if (query.isLinkElement(targetNode) &&
-                    query.hasNodeAttributeWithValue(targetNode, 'href')) {
-                    var url = String(query.getAttributeValueFromNode(targetNode, 'href'));
-
-                    if (0 === url.indexOf('#')) {
-                        return false;
-                    }
-
-                    if (startsUrlWithTrackerUrl(url)) {
-                        return true;
-                    }
-
-                    if (!content.isUrlToCurrentDomain(url)) {
-                        return false;
-                    }
-
-                    var block = content.buildContentBlock(contentBlock);
-
-                    if (!block) {
-                        return;
-                    }
-
-                    var contentName   = block.name;
-                    var contentPiece  = block.piece;
-                    var contentTarget = block.target;
-
-                    if (!query.hasNodeAttributeWithValue(targetNode, content.CONTENT_TARGET_ATTR) || targetNode.wasContentTargetAttrReplaced) {
-                        // make sure we still track the correct content target when an interaction is happening
-                        targetNode.wasContentTargetAttrReplaced = true;
-                        contentTarget = content.toAbsoluteUrl(url);
-                        query.setAnyAttribute(targetNode, content.CONTENT_TARGET_ATTR, contentTarget);
-                    }
-
-                    var targetUrl = buildContentInteractionTrackingRedirectUrl(url, 'click', contentName, contentPiece, contentTarget);
```

Comment:
Is there a reason to keep the method `buildContentInteractionTrackingRedirectUrl` (and its tests)?

Reply:
fixed